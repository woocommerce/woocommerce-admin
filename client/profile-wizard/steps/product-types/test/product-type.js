/**
 * External dependencies
 */
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { createElement } from '@wordpress/element';

/**
 * Internal dependencies
 */
import ProductType from '../product-type';

const defaultProps = {
	annualPrice: 120,
	label: 'Product type label',
	description: 'Product type description',
	moreUrl: 'https://woocommerce.com/my-product-type',
	slug: 'my-product-type',
};

describe( 'ProductType', () => {
	test( 'should render the product type', () => {
		const { container } = render(
			<ProductType { ...defaultProps } isMonthlyPricing={ false } />
		);
		expect( container ).toMatchSnapshot();
	} );

	test( 'should render the product type with monthly prices', () => {
		const { container } = render(
			<ProductType { ...defaultProps } isMonthlyPricing={ true } />
		);
		expect( container ).toMatchSnapshot();
	} );

	test( 'should show Popover on click', () => {
		const { container } = render(
			<ProductType { ...defaultProps } isMonthlyPricing={ true } />
		);

		const infoButton = screen.getByLabelText(
			'Learn more about recommended free business features',
			{
				selector: 'button',
			}
		);

		userEvent.click( infoButton );

		const popover = container.querySelector( '.components-popover' );
		const learnMoreLink = popover.querySelector( 'a' );
		expect( popover ).not.toBeNull();
		expect( popover.textContent ).toBe(
			defaultProps.description + ' Learn more'
		);
		expect( learnMoreLink.href ).toBe( defaultProps.moreUrl );
	} );
} );
