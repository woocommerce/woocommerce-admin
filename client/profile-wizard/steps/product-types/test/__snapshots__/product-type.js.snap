// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductType should render the product type 1`] = `
<div>
  <div
    class="woocommerce-product-type"
  >
    <span
      class="woocommerce-product-type__label"
    >
      Product type label
    </span>
    <button
      aria-label="Learn more about recommended free business features"
      class="components-button is-tertiary"
      type="button"
    >
      <svg
        aria-hidden="true"
        focusable="false"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 3.2c-4.8 0-8.8 3.9-8.8 8.8 0 4.8 3.9 8.8 8.8 8.8 4.8 0 8.8-3.9 8.8-8.8 0-4.8-4-8.8-8.8-8.8zm0 16c-4 0-7.2-3.3-7.2-7.2C4.8 8 8 4.8 12 4.8s7.2 3.3 7.2 7.2c0 4-3.2 7.2-7.2 7.2zM11 17h2v-6h-2v6zm0-8h2V7h-2v2z"
        />
      </svg>
    </button>
    
    <span
      class="components-truncate components-text woocommerce-pill css-1g4vivm-View-Text-sx-Base em57xhy0"
      data-wp-c16t="true"
      data-wp-component="Text"
    >
      <span
        class="screen-reader-text"
      >
        This product type requires a paid extension.
We'll add this to a cart so that
you can purchase and install it later.
      </span>
      $120 per year
    </span>
  </div>
</div>
`;

exports[`ProductType should render the product type with monthly prices 1`] = `
<div>
  <div
    class="woocommerce-product-type"
  >
    <span
      class="woocommerce-product-type__label"
    >
      Product type label
    </span>
    <button
      aria-label="Learn more about recommended free business features"
      class="components-button is-tertiary"
      type="button"
    >
      <svg
        aria-hidden="true"
        focusable="false"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 3.2c-4.8 0-8.8 3.9-8.8 8.8 0 4.8 3.9 8.8 8.8 8.8 4.8 0 8.8-3.9 8.8-8.8 0-4.8-4-8.8-8.8-8.8zm0 16c-4 0-7.2-3.3-7.2-7.2C4.8 8 8 4.8 12 4.8s7.2 3.3 7.2 7.2c0 4-3.2 7.2-7.2 7.2zM11 17h2v-6h-2v6zm0-8h2V7h-2v2z"
        />
      </svg>
    </button>
    
    <span
      class="components-truncate components-text woocommerce-pill css-1g4vivm-View-Text-sx-Base em57xhy0"
      data-wp-c16t="true"
      data-wp-component="Text"
    >
      <span
        class="screen-reader-text"
      >
        This product type requires a paid extension.
We'll add this to a cart so that
you can purchase and install it later.
      </span>
      $10 per month
    </span>
  </div>
</div>
`;
